pandas/tests/frame/methods/test_set_axis.py

```diff
@@ -41,6 +41,14 @@ def test_set_axis_copy(self, obj, using_copy_on_write):
                     tm.shares_memory(result.iloc[:, i], obj.iloc[:, i])
                     for i in range(obj.shape[1])
                 )
+        else:
+            if obj.ndim == 1:
+                assert tm.shares_memory(result, obj)
+            else:
+                assert any(
+                    tm.shares_memory(result.iloc[:, i], obj.iloc[:, i])
+                    for i in range(obj.shape[1])
+                )
```

Comment:
here you've explicitly set `copy=True`, so I don't think you need `if-else`, just do the `# check we DID make a copy` check in both cases